src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferenceCache.cs

```diff
@@ -130,9 +131,19 @@ public SymbolInfo GetSymbolInfo(SyntaxNode node, CancellationToken cancellationT
         // walks the root and checks all identifier tokens.
         //
         // otherwise, we can use the text of the document to quickly find candidates and test those directly.
-        return this.SyntaxTreeIndex.ProbablyContainsEscapedIdentifier(identifier)
-            ? _identifierCache.GetOrAdd(identifier, identifier => FindMatchingIdentifierTokensFromTree(identifier, cancellationToken))
-            : _identifierCache.GetOrAdd(identifier, _ => FindMatchingIdentifierTokensFromText(identifier, cancellationToken));
+        if (this.SyntaxTreeIndex.ProbablyContainsEscapedIdentifier(identifier))
+        {
+            return _identifierCache.GetOrAdd(
+                identifier,
+                identifier => FindMatchingIdentifierTokensFromTree(identifier, cancellationToken));
+        }
+
+        return _identifierCache.GetOrAdd(
+            identifier,
+            identifier => FindMatchingTokensFromText(
+                identifier,
+                static (identifier, token, @this) => @this.IsMatch(identifier, token),
+                this, cancellationToken));
```

Comment:
this is almost hte same.  just the addition of hte lambdaA+args to call nito a unified helper here.